tests/unit/query_plan_accumulate_aggregate.cpp

```diff
@@ -609,3 +617,396 @@ TEST(QueryPlan, Unwind) {
     expected_y_it++;
   }
 }
+
+TEST_F(QueryPlanAggregateOps, WithDataDistinct) {
```

Comment:
I think it might make sense to create a parametrized tests. A good example can be found in `tests/unit/storage_v2_edge.cpp`, but please let sync about how to do this in order to avoid too much extra work. My idea is similar to what is done in the tests of edges: have two test cases: one that is run with `true`/`false` parameters and another one that is independent from the value of distinct.